src/server/ftpserver/options.rs

```diff
@@ -216,3 +216,42 @@ impl Default for Shutdown {
         }
     }
 }
+
+#[derive(Debug, Clone)]
+/// Variants for failed logins protection policy
+pub enum FailedLoginsPolicy {
+    /// User plus source IP address locking
+    SourceUserLock(FailedLoginsPenalty),
+    /// Source IP locking
+    SourceIPLock(FailedLoginsPenalty),
+    /// Lock the user
+    UserLock(FailedLoginsPenalty),
```

Comment:
Sorry to be on about naming again but since it's the public API I deem it important. Is `Blocking` not a better term? The account gets `locked` but the user or IP gets `blocked`?

How about:

`BlockUser`, `BlockIP`, `BlockUserAndIP`?

Reply:
I welcome the naming suggestions. Sometimes it's hard to come up with good naming. I will change this.